core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -680,12 +680,15 @@ abstract class AbstractFetcherThread(name: String,
        */
       val offsetAndEpoch = leader.fetchEarliestOffset(topicPartition, currentLeaderEpoch)
       val leaderStartOffset = offsetAndEpoch.offset
-      warn(s"Reset fetch offset for partition $topicPartition from $replicaEndOffset to current " +
-        s"leader's start offset $leaderStartOffset")
       val offsetToFetch = Math.max(leaderStartOffset, replicaEndOffset)
+      info(s"Reset fetch offset for partition $topicPartition from $replicaEndOffset to " +
```

Comment:
If `leaderStartOffset <= replicaEndOffset` the fetch offset shouldn't change since it should already be set to `replicaEndOffset` no? In this case should we log it at all?

Reply:
As discussed with kamalcph, if came here, there may be an unclean leader election occurs. This is also a kind of abnormal case. I think it may still need to be logged.

Reply:
Makes sense on the benefit from logging.